### PR TITLE
Task: 198179 Identify AG form - step 1to5

### PR DESCRIPTION
**Pull request by swapnilphalke03**
Identify AG form - step 1to5

https://github.com/tekdi/eg-website/assets/133649148/40b7e80a-e2bd-463d-94c9-52bf2e02c065

### CODE REVIEW

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 1]
```diff
@@ -0,0 +1,1074 @@
...
+  const { image, takeScreenshot } = useScreenshot();
+  const getImage = () => takeScreenshot({ ref });
+  const downloadImage = () => {
+    var FileSaver = require("file-saver");
+    FileSaver.saveAs(`${image}`, "image.png");
+  };
+
+  React.useEffect(() => {
+    getImage();
+  }, [page, credentials]);
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:132]: Remove this not required here.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 2]
```diff
@@ -0,0 +1,1074 @@
...
+
+  
+
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:135]: Run yarn format for code indentation

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 3]
```diff
@@ -0,0 +1,1074 @@
...
+    },
+    experience: {
+      items: {
+        experience_in_years: { "ui:widget": CustomR },
+        related_to_teaching: {
+          "ui:widget": RadioBtn,
+        },
+      },
+    },
+  };
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:202]: Please check which parameter you are using for create ag and remove other.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 4]
```diff
@@ -0,0 +1,1074 @@
...
+    }
+
+    if (schema?.properties?.device_ownership) {
+      if (formData?.device_ownership == "no") {
+        setAlert(t("YOU_NOT_ELIGIBLE"));
+      } else {
+        setAlert();
+      }
+    }
+  }, [page]);
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:338]: Please check which core is required for create ag and remove other code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 5]
```diff
@@ -0,0 +1,1074 @@
...
+  React.useEffect(() => {
+    if (schema1.type === "step") {
+      const properties = schema1.properties;
+      const newSteps = Object.keys(properties);
+    setPage(newSteps[0]);
+    setSchema(properties[newSteps[0]]);
+      setPages(newSteps);
+      let minYear = moment().subtract("years", 50);
+      let maxYear = moment().subtract("years", 18);
+      setYearsRange([minYear.year(), maxYear.year()]);
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:349]: Please check which core is required for create ag and remove other code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 6]
```diff
@@ -0,0 +1,1074 @@
...
+      if (formData.experience?.length > 0) {
+        setSubmitBtn(t("NEXT"));
+        setAddBtn(t("ADD_EXPERIENCE"));
+      } else {
+        setSubmitBtn(t("NO"));``
+        setAddBtn(t("YES"));
+      }
+    } else {
+      setSubmitBtn(t("NEXT"));
+    }
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:373]: Please check which core is required for create ag and remove other code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 7]
```diff
@@ -0,0 +1,1074 @@
...
+  };
+
+  React.useEffect(() => {
+    updateBtnText();
+  }, [formData, page, lang]);
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:378]: Remove not required.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 8]
```diff
@@ -0,0 +1,1074 @@
...
+    ["vo_experience", "experience"].forEach((keyex) => {
+      data?.[keyex]?.map((item, index) => {
+        ["role_title", "organization", "description"].forEach((key) => {
+          if (item?.[key] && !item?.[key]?.match(/^[a-zA-Z ]*$/g)) {
+            errors[keyex][index]?.[key]?.addError(
+              `${t("REQUIRED_MESSAGE")} ${t(schema?.properties?.[key]?.title)}`
+            );
+          }
+        });
+      });
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:457]: Please check which core is required for create ag and remove other code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 9]
```diff
@@ -0,0 +1,1074 @@
...
+        if (result.isUserExist) {
+          const newErrors = {
+            mobile: {
+              __errors: [t("MOBILE_NUMBER_ALREADY_EXISTS")],
+            },
+          };
+          setErrors(newErrors);
+        }
+      }
+    }
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:579]: Please confirm this validation is required

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 10]
```diff
@@ -0,0 +1,1074 @@
...
+            valueIndex = schema?.properties?.qualification?.enum[index];
+          }
+        });
+        if (valueIndex !== "" && data.qualification == valueIndex) {
+          setAlert(t("YOU_NOT_ELIGIBLE"));
+        } else {
+          setAlert();
+        }
+      }
+    }
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:610]: Remove this is not required

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 11]
```diff
@@ -0,0 +1,1074 @@
...
+
+    if (id === "root_device_ownership") {
+      if (schema?.properties?.device_ownership) {
+        if (data?.device_ownership == "no") {
+          setAlert(t("YOU_NOT_ELIGIBLE"));
+        } else {
+          setAlert();
+        }
+      }
+    }
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:620]: Please check which core is required for create ag and remove other code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 12]
```diff
@@ -0,0 +1,1074 @@
...
+        <VStack py={6} px={4} mb={5} space="6">
+          {/* <Box p="10">
+            <Steper
+              steps={[
+                { value: "6", label: t("BASIC_DETAILS") },
+                { value: "3", label: t("WORK_DETAILS") },
+                { value: "4", label: t("OTHER_DETAILS") },
+              ]}
+              progress={page === "upload" ? 13 : page}
+            />
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:746]: Remove commented code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 13]
```diff
@@ -0,0 +1,1074 @@
...
+          </Box> */}
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:816]: Remove commented code.

File: apps/front-end/src/pages/front-end/ag-form/Agform.js [hunk 14]
```diff
@@ -0,0 +1,1074 @@
...
+                    navigate(0);
+                  }}
+                >
+                  {t("LOGIN")}
+                </Button>
+              </HStack>
+            </VStack>
+          </Modal.Body>
+        </Modal.Content>
+      </Modal>
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/Agform.js:1071]: Is this modal is not required remove

File: apps/front-end/src/pages/front-end/ag-form/parts/Schema.js [hunk 15]
```diff
@@ -0,0 +1,154 @@
...
+    //           "prerak_referral",
+    //           "ngo_referral",
+    //           "advertisements",
+    //           "dropouts",
+    //           "old_prerak",
+    //           "other",
+    //         ],
+    //       },
+    //     },
+    //   },
```
Comment (sagarkoshti1990) [inline apps/front-end/src/pages/front-end/ag-form/parts/Schema.js:151]: Remove commented code.

**Comment by swapnilphalke03** [comment]
Removed all unwanted Code and console logs.
Screenshot from 2023-05-19 10-00-50

**Pull request merged**